infra/feast-operator/internal/controller/services/util.go

```diff
@@ -45,6 +45,14 @@ func shouldCreatePvc(featureStore *feastdevv1alpha1.FeatureStore, feastType Feas
 func ApplyDefaultsToStatus(cr *feastdevv1alpha1.FeatureStore) {
 	cr.Status.FeastVersion = feastversion.FeastVersion
 	applied := cr.Spec.DeepCopy()
+
+	if applied.AuthConfig == nil {
+		applied.AuthConfig = &feastdevv1alpha1.AuthConfig{}
+	}
+	if applied.AuthConfig.KubernetesAuth != nil && applied.AuthConfig.KubernetesAuth.Roles == nil {
+		applied.AuthConfig.KubernetesAuth.Roles = []string{}
```

Comment:
curious why we would default to an empty slice here?

Reply:
just thinking it could help in range loops, but I see it's not causing any harms. removing it, thanks